lucene/expressions/src/java/org/apache/lucene/expressions/SimpleBindings.java

```diff
@@ -96,24 +90,51 @@ public DoubleValuesSource getDoubleValuesSource(String name) {
       case SCORE:
         return DoubleValuesSource.SCORES;
       default:
-        throw new UnsupportedOperationException(); 
+        throw new UnsupportedOperationException();
     }
   }
   
-  /** 
-   * Traverses the graph of bindings, checking there are no cycles or missing references 
-   * @throws IllegalArgumentException if the bindings is inconsistent 
+  @Override
+  public DoubleValuesSource getDoubleValuesSource(String name) {
+    if (map.containsKey(name) == false) {
```

Comment:
Is `containsKey` boolean check more clear than `get` with a null check? I think the latter is going to be more efficient because it's only a single map operation, but I guess this way might be better for the JVM's escape analysis?

Reply:
I'm pretty sure this won't be in the hot path - it's part of query setup, not query execution - and I think it reads more clearly.